interlocking/interlockingcontroller/overlapcontroller.py

```diff
@@ -1,89 +1,100 @@
+from interlocking.model import OccupancyState
+
+
 class OverlapController(object):
 
     def __init__(self, interlocking, track_controller, point_controller):
         self.interlocking = interlocking
         self.track_controller = track_controller
         self.point_controller = point_controller
 
-    def reserve_overlap_of_route(self, route):
-        overlap = self.get_first_reservable_overlap(route)
+    def reserve_overlap_of_route(self, route, train_id: str):
+        overlap = self.get_first_reservable_overlap(route, train_id)
```

Comment:
Not specific to this pr, but why do you only reserve the first overlap you find?

Reply:
This is for sure a point where an optimised algorithm could increase the performance of the logic. For example by predicting on-coming routes and choose clever overlaps to not blocking them.